jupyter/radon/radon.ipynb

```diff
@@ -0,0 +1,1961 @@
+    "When developing a visualization, the initial choice of geom might\n",
+    "not work properly with additional layers.\n",
+    "Plot building, like model building, is an iterative process.\n",
+    "A good resource is [Data Visualization with Plotnine](https://f0nzie.github.io/rmarkdown-python-plotnine/),\n",
+    "a Python translation of [R for Data Science](https://r4ds.had.co.nz/).\n",
+    "\n",
+    "**First questions: amount of data, variable of interest**\n",
+    "\n",
+    "How much data is there for Minnesota?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "601ad455",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f'number of houses: {len(mn_radon)}')\n",
+    "print(f'number of counties: {len(mn_counties)}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "130b19ac",
+   "metadata": {},
+   "source": [
+    "The goal of our analysis is to estimate home radon levels; therefore the outcome variable of interest is `log_radon`.\n",
+    "We use the [pandas.DataFrame.describe](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.describe.html) function to get summary statistics over the observed outcome `log_radon`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bfcfd10d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f'log_radon summary statistics\\n{mn_radon[\"log_radon\"].describe()}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "07cf3160",
+   "metadata": {},
+   "source": [
+    "**Relationship between radon and floor**\n",
+    "\n",
+    "As the radon pathways diagram shows, radon comes from the soil, therefore the floor level on which the measurement was taken should be a good predictor of the observed radon level. This is coded as \"0\" for basement and \"1\" for ground floor level.  Most of the observations in the survey were taken on the basement level."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "00d08f36",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pct_1 = round((mn_radon.floor.sum() / len(mn_radon) * 100))\n",
+    "pct_0= round(100 - pct_1)\n",
+    "print(f'floor 0: {pct_0}%\\nfloor 1: {pct_1}%')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "45fbbe7e",
+   "metadata": {},
+   "source": [
+    "The [plotnine geom_histogram](https://plotnine.readthedocs.io/en/stable/generated/plotnine.geoms.geom_histogram.html) provides a visualization of the distribution of home radon measurements in the survey dataset.\n",
+    "Here we overlay two per-floor histograms.\n",
+    "They clearly show the different amounts of observation (y axis), but the trend in log radon levels (x axis) is unclear."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7473e74e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# overlay histograms\n",
```

Comment:
don't see an obvious way to do this - tried stuff that might work in R with ggplo2, but doesn't for plotnine.
might be because R does all sorts of crazy lazy eval with names that Python doesn't.

Reply:
The following code produces a nearly equivalent histogram but with labels:

```python
# overlay histograms
(p9.ggplot(data=mn_radon, mapping=p9.aes(x='log_radon', color='factor(floor)', fill='factor(floor)'))
    + p9.geom_histogram(alpha=0.7, binwidth=0.2)
    + p9.theme(figure_size=(6,4))
    + p9.scale_color_manual(['darkgreen','blue'])
    + p9.scale_fill_manual(['orange','violet'])
)
```
I say nearly equivalent because this has slightly different properties in the overlapping section so it isn't a perfect visual match. An interesting puzzle to figure out how to do it, though!

Reply:
awesome!!! 
will add prose explaining - also add some more plotnine refs - this one, for example: https://datacarpentry.org/python-ecology-lesson/07-visualization-ggplot-python/

plotnine doc is like the unix man pages - only helpful if you already know the system.

Reply:
I started with the observation that a lot of the plotnine examples have legends, and that usually corresponds to the use of the `factor` keyword. Went down the rabbit hole from there and tried various things, then control-F'd for "fill" on their API doc to find `scale_fill_manual` and friends.

Aka, not very easily. Definitely agree about the man-page like nature of it.

Reply:
YATA!